x-pack/test_serverless/functional/test_suites/observability/dataset_quality/dataset_quality_flyout.ts

```diff
@@ -113,5 +121,164 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
+
+      await PageObjects.observabilityLogsExplorer.navigateTo();
+
+      // Add initial integrations
+      await PageObjects.observabilityLogsExplorer.setupInitialIntegrations();
+
+      // Index 10 logs for `logs-apache.access` dataset
+      await synthtrace.index(
+        getLogsForDataset({ to, count: 10, dataset: apacheAccessDatasetName })
+      );
+
+      await PageObjects.datasetQuality.navigateTo();
+
+      await PageObjects.datasetQuality.openDatasetFlyout(apacheAccessDatasetHumanName);
+      await PageObjects.datasetQuality.openIntegrationActionsMenu();
+
+      const actions = await Promise.all(
+        Object.values(integrationActions).map((action) =>
+          PageObjects.datasetQuality.getIntegrationActionButtonByAction(action)
+        )
+      );
+
+      expect(actions.length).to.eql(3);
+    });
+
+    it('Integration dashboard action hidden for integrations without dashboards', async () => {
+      const bitbucketDatasetName = 'atlassian_bitbucket.audit';
+      const bitbucketDatasetHumanName = 'Bitbucket Audit Logs';
+
+      await PageObjects.observabilityLogsExplorer.navigateTo();
+
+      // Add initial integrations
+      await PageObjects.observabilityLogsExplorer.installPackage({
+        name: 'atlassian_bitbucket',
+        version: '1.14.0',
+      });
+
+      // Index 10 logs for `atlassian_bitbucket.audit` dataset
+      await synthtrace.index(getLogsForDataset({ to, count: 10, dataset: bitbucketDatasetName }));
+
+      await PageObjects.datasetQuality.navigateTo();
+
+      await PageObjects.datasetQuality.openDatasetFlyout(bitbucketDatasetHumanName);
+      await PageObjects.datasetQuality.openIntegrationActionsMenu();
+
+      await testSubjects.missingOrFail(
+        PageObjects.datasetQuality.testSubjectSelectors.datasetQualityFlyoutIntegrationAction(
+          integrationActions.viewDashboards
+        )
+      );
+    });
+
+    it('Integration overview action should navigate to the integration overview page', async () => {
+      const bitbucketDatasetName = 'atlassian_bitbucket.audit';
+      const bitbucketDatasetHumanName = 'Bitbucket Audit Logs';
+
+      await PageObjects.observabilityLogsExplorer.navigateTo();
+
+      // Add initial integrations
+      await PageObjects.observabilityLogsExplorer.installPackage({
+        name: 'atlassian_bitbucket',
+        version: '1.14.0',
+      });
+
+      // Index 10 logs for `atlassian_bitbucket.audit` dataset
+      await synthtrace.index(getLogsForDataset({ to, count: 10, dataset: bitbucketDatasetName }));
+
+      await PageObjects.datasetQuality.navigateTo();
+
+      await PageObjects.datasetQuality.openDatasetFlyout(bitbucketDatasetHumanName);
+      await PageObjects.datasetQuality.openIntegrationActionsMenu();
+
+      const action = await PageObjects.datasetQuality.getIntegrationActionButtonByAction(
+        integrationActions.overview
+      );
+
+      await action.click();
+
+      await retry.tryForTime(5000, async () => {
```

Comment:
Could we extract this into a helper that checks users have been redirected correctly?